src/Features/Core/Portable/EncapsulateField/AbstractEncapsulateFieldService.cs

```diff
@@ -244,27 +245,35 @@ private async Task<Result> EncapsulateFieldAsync(IFieldSymbol field, Document do
             if (field.IsReadOnly)
             {
                 // Inside the constructor we want to rename references the field to the final field name.
-                var constructorSyntaxes = GetConstructorNodes(field.ContainingType).ToSet();
-                if (finalFieldName != field.Name && constructorSyntaxes.Count > 0)
+                var constructorLocations = GetConstructorLocations(field.ContainingType);
+                if (finalFieldName != field.Name && constructorLocations.Count > 0)
                 {
-                    solution = await Renamer.RenameSymbolAsync(solution,
-                        SymbolAndProjectId.Create(field, projectId),
-                        finalFieldName, solution.Options,
-                        location => constructorSyntaxes.Any(c => c.Span.IntersectsWith(location.SourceSpan)),
-                        cancellationToken: cancellationToken).ConfigureAwait(false);
+                    var initialLocations = await Renamer.GetRenameLocationsAsync(
+                        solution, SymbolAndProjectId.Create(field, projectId), solution.Options, cancellationToken).ConfigureAwait(false);
+
+                    var insideLocations = initialLocations.Filter(
+                        location => IntersectsWithAny(location, constructorLocations));
+
+                    solution = await Renamer.RenameAsync(
+                        insideLocations, finalFieldName, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
the overload that has the rename engine filter is gone.  now, features can jut ask for the locations that will be renamed, filter them themselves, then ask to rename the filtered set.  

Both GetRenameLocationsAsync and RenameAsync will be OOPable in an upcoming PR.

Reply:
Nit: consider extracting out the common code into a shared helper parameterized by `locationFilter`, say

```csharp
                  static async Task<Solution> RenameSymbolAsync(locationFilter)
                  {
                    var initialLocations = await Renamer.GetRenameLocationsAsync(
                        solution, SymbolAndProjectId.Create(field, projectId), solution.Options, cancellationToken).ConfigureAwait(false);

                    var insideLocations = initialLocations.Filter(locationFilter);

                    return await Renamer.RenameAsync(
                        insideLocations, finalFieldName, cancellationToken: cancellationToken).ConfigureAwait(false);
                  }

Reply:
will do in next PR.  thanks!